app/controllers/hq/orphans_controller.rb

```diff
@@ -1,6 +1,61 @@
 class Hq::OrphansController < HqController
 
   def show
+    load_orphan
+  end 
+
+  def index
+    @orphans = Orphan.paginate(:page => params[:page])
+  end
+
+  def edit
+    load_orphan
+    load_associations
+  end
+
+  def update
+    load_orphan
+    build_orphan
+    save_orphan or re_render 'edit'
+  end
+
+private
+  def load_orphan
     @orphan = Orphan.find(params[:id])
   end
+
+  def orphan_params
+    params.require(:orphan)
+      .permit(:contact_number, :date_of_birth, :family_name, :father_deceased, 
+              :health_status, :schooling_status, :father_given_name, 
+              :father_is_martyr, :gender, :minor_siblings_count, :mother_alive, 
+              :mother_name, :name, :orphan_list_id, :orphan_sponsorship_status_id, 
+              :orphan_status_id, :priority, :sponsored_by_another_org, :sponsored_minor_siblings_count, 
+              :minor_siblings_count, :sponsored_minor_siblings_count, 
+              original_address: [:id, :city, :province_id, :street, :neighborhood],
+              current_address: [:id, :city, :province_id, :street, :neighborhood])
+  end
+
+  def build_orphan
+    @orphan ||= Orphan.new
+    @orphan.attributes = orphan_params if params[:orphan]
+  end
```

Comment:
I've never been a fan of this method in other HQ controllers. It manages to squeeze two branches into two lines of code, which is a good indication that it does too much. What are the benefits of using it instead of explicit assignments in controller actions? Here in the orphans controller, it is particularly redundant as it is only called from `update`, which means that there is no need for the conditionals: both `@orphan` and `params[:orphan]` will always exist by the time this method gets called. It looks like we can save 4 LOC and eliminate branching by replacing `build_orphan` with `@orphan.attributes = orphan_params` in the `update` action and deleting this method. What do you think?